localstack/services/dynamodb/provider.py

```diff
@@ -615,19 +620,33 @@ def list_tables(
         exclusive_start_table_name: TableName = None,
         limit: ListTablesInputLimit = None,
     ) -> ListTablesOutput:
-        return self.forward_request(context)
+        response = self.forward_request(context)
+
+        # Add replicated tables
+        replicas = get_store(context.account_id, context.region).REPLICA_UPDATES
+        for replicated_table, replications in replicas.items():
+            for original_region, replicated_regions in replications.items():
+                if context.region in replicated_regions:
+                    response["TableNames"].append(replicated_table)
+
+        return response
```

Comment:
Replicated tables must be listed when querying in original region as well as replicated regions.